tensorflow/python/framework/python_op_gen.cc

```diff
@@ -1090,4 +1245,19 @@ string GetPythonWrappers(const char* op_list_buf, size_t op_list_len) {
   return GetPythonOpsImpl(ops, api_def_map, {});
 }
 
+string GetArgAnnotation(const auto& arg, std::unordered_map<string, string>& type_annotations) {
+  if (type_annotations.find(arg.type_attr()) != type_annotations.end()) {
+    // Get the correct TypeVar if arg maps to an attr
+    return "_ops.Tensor[" + type_annotations[arg.type_attr()] + "]";
+  } else {
+    // Get the dtype of the Tensor
+    const string py_dtype = python_op_gen_internal::DataTypeToPython(arg.type(), "_dtypes.");
+    if (dtype_type.find(py_dtype) != dtype_type.end()) {
```

Comment:
I think we expect to always find a matching dtype, correct?

Reply:
Yeah, we will always find a match now that we added support for `dtypes.half`.